src/model.py

```
@@ -135,38 +139,74 @@ class BOWFeatures(BaseEstimator, TransformerMixin):
+                                will be removed from the dictionary
         
         """
         self.bow = bow
+        self.use_passed_bow = use_passed_bow
+        self.include_proper_nouns = include_proper_nouns
+        self.max_occurences = max_occurences
+        self.min_occurences = min_occurences
         
 
     def transform(self, X):
         """vectorizes a row of text data using the bag of words"""
-
         vectorizer = CountVectorizer(analyzer='word',token_pattern=r'\w{1,}',
                                 ngram_range=(1, 1), vocabulary=self.bow)
+        text_frame = _convert_to_dataframe(X)
+        
+         #-----------lemmatize before vectorizing--------------------------------------------   
+        text_frame['text'] = text_frame['text'].apply(lambda text: BookText(rawtext=text)
+                                                      .clean(lemmatize=True, deromanize=True)
+                                                      .text)
+        #------------------------------------------------------------------------------------
         
+        X = text_frame['text'].values
         XX = vectorizer.fit_transform(X)
         
         return XX
         
 
     def fit(self, X, y=None):
-        """Fit DataFrame by building bag of words"""
+        """Fit DataFrame by building bag of words
+        
+        Uses the passed bow if the flag self.use_passed_bow is True
+        
+        """
         text_frame = _convert_to_dataframe(X)
-        if self.bow =={}:
-            self.build_bow(corpus = text_frame['text'].values)
+        
+        if not self.use_passed_bow:
+            self.build_bow(corpus = text_frame['text'])
+            
+        else:
+            if self.bow =={}:
+                print("Looks like you forgot to pass a Bag of words.")
+                print("If you want to build a bag instead, set use_passed_bag=False .")
```

This should probably raise a `ValueError` instead of printing out the warning

---

OK. That's reasonable.